packages/theme/src/cli/services/update/run.ts

```diff
@@ -0,0 +1,178 @@
+  get theme(): Theme
+  set theme(value: Theme)
+  get scriptContent(): string
+  set scriptContent(value: string)
+}
+
+interface Options {
+  script: string
+  'source-theme': string
+  'target-theme': string
+}
+
+// 5 minutes
+const UPDATER_TIMEOUT = 5 * 60 * 1000
+
+export async function run(session: AdminSession, options: Options) {
+  const ctx = await renderTasks<Context>([
+    task(`Checking your theme update script`, (ctx) => check(ctx, options)),
+    task(`Starting your theme update`, (ctx) => triggerUpdater(ctx, session, options)),
+    task(`Updating your theme`, (ctx) => waitForUpdater(ctx, session)),
+  ])
+
+  renderSuccess({
+    body: 'Your theme has been updated.',
+    nextSteps: [
+      [
+        {
+          link: {
+            label: 'Explore the updated theme in the code editor',
+            url: codeEditorUrl(ctx.theme, session),
+          },
+        },
+      ],
+      [
+        {
+          link: {
+            label: 'Explore the updated theme in the theme editor',
+            url: themeEditorUrl(ctx.theme, session),
+          },
+        },
+      ],
+      [
+        {
+          link: {
+            label: 'Preview the updated theme',
+            url: themePreviewUrl(ctx.theme, session),
+          },
+        },
+      ],
+    ],
+  })
+}
+
+async function check(ctx: Context, options: Options) {
+  const {script} = options
+
+  await sleep(1)
+  await checkScript(script)
+
+  ctx.scriptContent = await readFile(joinPath(script))
+}
+
+async function triggerUpdater(ctx: Context, session: AdminSession, options: Options) {
+  const {'source-theme': source, 'target-theme': target} = options
+  const store = session.storeFqdn
+
+  const themes = await fetchStoreThemes(session)
+
+  const sourceThemeId = findThemeByIdentifier(store, themes, source).id
+  const targetThemeId = findThemeByIdentifier(store, themes, target).id
+
+  // This is a false positive, as `renderTasks` performs tasks sequentially.
+  // eslint-disable-next-line require-atomic-updates
+  ctx.theme = await triggerUpdaterAPI(session, ctx.scriptContent, sourceThemeId, targetThemeId)
+}
+
+async function waitForUpdater(ctx: Context, session: AdminSession) {
+  const startTime = Date.now()
+
+  /**
+   * Generally, it's a good practice to take full advantage of the parallelization
+   * benefits of async/await.
+   *
+   * However, in the context of polling an API, each iteration must wait for the
+   * response, before the next polling.
+   */
+  // eslint-disable-next-line no-await-in-loop
+  while (await isUpdaterIsProgress(ctx, session, startTime)) {
+    // retry
+  }
+}
+
+async function isUpdaterIsProgress(ctx: Context, session: AdminSession, startTime: number) {
+  const theme = await fetchTheme(ctx.theme.id, session)
+
+  if (!theme) {
+    throw new AbortError(
+      'The `update_extension.json` script could not be executed due to a runtime issue.',
+      'Please check the `update_extension.json` script and retry.',
```

Comment:
This error message should be generic. The update process could fail for many different reasons. Additionally in majority of cases there won't be any update_extension passed in from theme-partners. 
So something a long the lines of "Theme updated failed...". We can probably use the same generic error that we use in OSW on theme index.

Reply:
> The update process could fail for many different reasons
Indeed!

> Additionally in majority of cases there won't be any update_extension passed in from theme-partners.

I'm not so sure about this part.  Since the script is still in development, it may fail due to issues in the work-in-progress extension script.

Also, in this context, there are only three elements the partner can control: the update extension script, the source theme, and the target theme.

Generally, error messages should display the cause of the error and the potential solution to the problem. With this in mind, I believe we could include information about the source and target themes. However, I don't think making the message more generic would be beneficial for our partners. What do you think about that?